test/e2e/persistent_volumes.go

```diff
@@ -327,241 +411,396 @@ func createWaitAndDeletePod(f *framework.Framework, c *client.Client, ns string,
+	// Testing configurations of single a PV/PVC pair attached to a GCE PD
+	framework.KubeDescribe("PV:GCEPD", func() {
+
+		var (
+			diskName  string
+			err       error
+			pv        *api.PersistentVolume
+			pvc       *api.PersistentVolumeClaim
+			clientPod *api.Pod
+			pvConfig  persistentVolumeConfig
+		)
+
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("gce")
+			if diskName == "" {
+				diskName, err = createPDWithRetry()
+				Expect(err).NotTo(HaveOccurred())
+				pvConfig = persistentVolumeConfig{
+					namePrefix: "gce-",
+					pvSource: api.PersistentVolumeSource{
+						GCEPersistentDisk: &api.GCEPersistentDiskVolumeSource{
+							PDName:   diskName,
+							FSType:   "ext3",
+							ReadOnly: false,
+						},
+					},
+					prebind: nil,
+				}
+			}
+		})
+
+		AfterEach(func() {
+			if c != nil {
+				if clientPod != nil && len(clientPod.Name) > 0 {
+					_, err := c.Pods(ns).Get(clientPod.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+					        framework.Logf("AfterEach: deleting client Pod %v", clientPod.Name)
+        					err = c.Pods(ns).Delete(clientPod.Name, nil)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted client Pod %v", clientPod.Name)
+					}
+				}
+				clientPod = nil
+				if pvc != nil && len(pvc.Name) > 0 {
+					_, err := c.PersistentVolumeClaims(ns).Get(pvc.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleting PVC %v", pvc.Name)
+						err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted PVC %v", pvc.Name)
+					}
+				}
+				pvc = nil
+				if pv != nil && len(pv.Name) > 0 {
+					_, err := c.PersistentVolumes().Get(pv.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleting PV %v", pv.Name)
+						err := c.PersistentVolumes().Delete(pv.Name)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted PV %v", pv.Name)
+					}
+				}
+				pv = nil
+			}
+		})
+
+		AddCleanupAction(func() {
+			if len(diskName) > 0 {
+				deletePDWithRetry(diskName)
+ 			}
+		})
+
+		// Attach a persistent disk to a pod using a PVC.
+		// Delete the PVC and then the pod.  Expect the pod to succeed in unmounting and detaching PD on delete.
+		It("should test that deleting a PVC before the pod does does not cause unmounting and detaching the PD" +
+			"to fail on pod deletion", func() {
+			By("Creating the PV and PVC")
+			pv, pvc = createPVPVC(c, pvConfig, ns, false)
+
+			By("Creating the Client Pod")
+			clientPod := makePod(ns, pvc.Name)
+			clientPod, err = c.Pods(ns).Create(clientPod)
+			framework.WaitForPodRunningInNamespace(c, clientPod)
+
+			By("Deleting the Claim")
+			err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Sorry I've only been skimming this pr, but where is the verification that the mount (on host I presume) still exists? Ditto for deleting the pv in the next It() below.